doc/platform/brcm_pdk_pddf.md

```diff
@@ -0,0 +1,1170 @@
+
+# Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 05/27/2019  |  Systems Infra Team     | Initial version                   |
+| 0.2 | 06/06/2019  |  Systems Infra Team     | Incorporated feedback             |
+
+# About this Manual
+Platform Driver Development Framework (PDDF) is part of SONiC Platform Development Kit (PDK) which optimizes the platform development. PDK consists of 
+ - PDDF (Platform Driver Development Framework): For optimized data-driven platform driver and SONiC plugin development
+ - PDE (Platform Development Environment): For optimized build and test of platform and SAI code
+
+PDE details are covered in another document. This document describes Platform Driver Development Framework (PDDF) which can be used as an alternative to the existing manually-written SONiC platform driver framework. It enables platform vendors to rapidly develop the device specific custom drivers and SONiC user space python plugins, using a data-driven architecture, to manage platform devices like Fan, PSUs, LEDs, Optics, System EEPROM, etc., and validate a platform on SONiC. 
+
+# Scope
+This document describes the high level design details of PDDF and its components. The PDDF consists of generic device drivers and user space platform API plugins which use the per platform specific data in the JSON descriptor files. This document describes the interaction between all the components and the tools used to support these drivers and plugins.  
+
+
+# Definition/Abbreviation
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+| ODM                      | Original Design Manufacturer        |
+| OEM                      | Original Equipment Manufacturer        |
+| PDDF                      | Platform Driver Development Framework         |
+| PDE                       | Platform Development Environment              |
+| PDK                      | Platform Development Kit              |
+| SAI                      | Switch Abstraction Interface |
+| PSU                      | Power Supply Unit |
+| I2C                      | Inter-integrated Circuit communication protocol |
+| SysFS                    | Virtual File System provided by the Linux Kernel |
+
+
+## 1 Requirements Overview
+SONiC OS is portable across different network devices with supported ASIC via Switch Abstraction Interface (SAI). These devices primarily differ in the way various device specific hardware components are accessed, and thus require custom device drivers and python plugins. Each platform vendor implements these custom device drivers and plugins. The feature requirement is to support a SONiC platform driver development framework to enable rapid development of custom device drivers and plugins.
+
```

Comment:
Currently SONiC is the process to replace platform plugin with a newly designed platform APIs,  the target is to switch to  new platform API in 201908 release, so should also considering to support platform API for the coming transform.  The design doc of new platform API is here:  https://github.com/Azure/SONiC/pull/285  the new platform base class is defined here: https://github.com/Azure/sonic-platform-common/tree/master/sonic_platform_base

Reply:
Hi Kebo Liu,

Yes, we are aware of the new platform API in August, 2019 release. We plan to support new platform base class in PDDF.

-Fuzail

Reply:
@fk410167: Do you have plans to update this document to reflect the new platform API? Vendors are currently in the process of migrating over to the new API. At this point in time, this document should reflect the new API rather than the old plugins.

Reply:
Yes we have already started to move the whole framework to use 2.0 platform APIs. We will modify the HLD accordingly.